### PR TITLE
Add module: Clustalo align

### LINKED ISSUE #3731: new module: clustalo/align
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by luisas**
Add module: Clustalo align

Takes as input a fasta and optionally a guide tree. Computes an MSA with clustalo.

## PR checklist

Closes #3731

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/clustalo/align/meta.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+---
+# yaml-language-server: $schema=https://raw.githubusercontent.com/nf-core/modules/master/modules/yaml-schema.json
```
Comment (sateeshperi) [inline modules/nf-core/clustalo/align/meta.yml:2]:
```suggestion
```

File: tests/modules/nf-core/clustalo/align/main.nf [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+
+workflow test_clustalo_align_with_tree {
+    
+    input = [
+        [ id:'test' ], // meta map
+        file(params.test_data['sarscov2']['illumina']['contigs_fasta'], checkIfExists: true)
+    ]
+
+    ch_tree = CLUSTALO_GUIDETREE ( input ).tree
```
Comment (sateeshperi) [inline tests/modules/nf-core/clustalo/align/main.nf:26]:
```suggestion
    ch_tree = CLUSTALO_GUIDETREE ( input ).tree
```
what is the purpose of this `.tree` at the end ?

File: modules/nf-core/clustalo/align/main.nf [hunk 3]
```diff
@@ -0,0 +1,50 @@
+
```
Comment (sateeshperi) [inline modules/nf-core/clustalo/align/main.nf:1]:
```suggestion
```

File: tests/modules/nf-core/clustalo/align/main.nf (hunk 2 above)
Comment (luisas) [reply tests/modules/nf-core/clustalo/align/main.nf:26]: This is to retrieve the right output channel from that module. The module CLUSTALO_GUIDETREE has 2 channel outputs, tree and versions. 

https://github.com/luisas/modules/blob/9a884757b561688e0b3ff8b55ff7eb4da25eef33/modules/nf-core/clustalo/guidetree/main.nf#L14-L15

File: modules/nf-core/clustalo/align/main.nf [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    tuple val(meta), path("*.aln"), emit: alignment
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def options_tree = tree ? "--guidetree-in=$tree" : ""
```
Comment (sateeshperi) [inline modules/nf-core/clustalo/align/main.nf:24]: is the `options_tree` used somewhere in the process or can they be supplied via args ?

Comment (luisas) [reply modules/nf-core/clustalo/align/main.nf]: You are right, I thought it was not possible but I just tested it and it works! 
I just updated the branch.
Thanks a lot :)

**Review by sateeshperi — APPROVED** [review]
lgtm... gr8 job!

**Pull request merged**